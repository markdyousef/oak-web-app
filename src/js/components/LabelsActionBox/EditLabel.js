// @flow
import React from 'react';
import styled from 'styled-components';
import Input from '../shared/Input';
import { SquareButton } from '../shared/Button';
import colors from '../../styles/colors';
import BackIcon from '../../icons/back';
import CheckIcon from '../../icons/checkmark';

const Grid = styled.div`
    display: flex;
    flex-wrap: wrap;
    width: calc(100% + 10px);
    flex-grow: 1;
    margin: 0px 0 0 -10px;
`;

const Label = styled.div`
    width: 30px;
    height: 33.5px;
    border-radius: 3px;
    cursor: pointer;
    margin: 5px;
    display: flex;
    align-items: center;
    justify-content: center;
    width: calc(100% * (1/5) - 10px - 1px);
    flex-grow: 1;
    margin: 10px 0 0 10px;
    & svg {
        fill: ${colors.white};
        position: relative;
        top: 1px;
    }
`;

const Section = styled.section`
    margin: 16px 0;
    & h5 {
        font-size: 12px;
        font-weight: lighter;
        color: ${colors.grey};
        font-family: 'proxima-nova'
    }
`;

const ActionSection = styled.section`
    border-top: 1px solid ${colors.lightGrey};
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 16px 20px 0px;
    margin-left: -20px;
    margin-right: -20px;
`;

const Back = styled.button`
    cursor: pointer;
    border: none;
    background-color: #fff;
    padding: 0;
    font-weight: normal;
    font-size: 13px;
    display: flex;
    align-items: center;
    & svg {
        margin-right: 6px;
        height: 13px;
        margin-left: -2px;
    }
`;

const Save = styled.button`
    cursor: pointer;
    border: none;
    background-color: #fff;
    padding: 0;
    font-size: 14px;
`;

const Delete = styled(Save)`
    color: #131517;
`;

type Props = {
    onCreate?: Function,
    onChange: Function,
    changePage: Function,
    labelName: ?string,
    selectedColor: string,
    labelColors: Array<string>
}

export default ({ ...props }:Props) => {
    const {
        changePage,
        labelName,
        onChange,
        labelColors,
        selectedColor,
        onCreate,
        onUpdate,
        onDelete
    } = props;
    return (
        <div>
            <Back onClick={changePage}>
                <BackIcon />
                Back
            </Back>
            <Section>
                <Input
                    title="NAME"
                    value={labelName || ''}
                    onChange={name => onChange({ name })}
                    placeholder="Label name"
                />
            </Section>
            <Section>
                <h5>SELECT COLOR: </h5>
                <Grid>
                    {labelColors.map((color) => {
                        const isSelected = (selectedColor === color);
                        return (
                            <Label
                                style={{ backgroundColor: color }}
                                key={color}
                                onClick={() => onChange({ color })}
                            >
                                {isSelected && <CheckIcon />}
                            </Label>
                        );
                    }
                    )}
                </Grid>
            </Section>
            <ActionSection>
                {onCreate &&
                    <Save onClick={onCreate}>
                        Save label
                    </Save>
                }
                {onDelete &&
                    <Delete onClick={onDelete}>
                        Delete
                    </Delete>
                }
                {onUpdate &&
                    <Save onClick={onUpdate}>
                        Savel label
                    </Save>
                }
            </ActionSection>
        </div>
    );
};
