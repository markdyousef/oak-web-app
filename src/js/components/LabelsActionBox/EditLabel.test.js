// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import EditLabel from './EditLabel';

it('renders correctly', () => {
    const props = {
        changePage: () => {},
        labelColors: [],
        labelName: '',
        onChange: () => {},
        onCreate: () => {},
        selectedColor: '#000'

    };
    const component = renderer.create(
        <EditLabel {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
