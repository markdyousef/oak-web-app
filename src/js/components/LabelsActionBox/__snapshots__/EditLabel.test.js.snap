// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    className="dOaeTO"
    onClick={[Function]}
  >
    <svg
      height="16"
      viewBox="0 0 16 16"
      width="16"
    >
      <g>
        <path
          d="M5.09,13.19L0.26,8.35C0.16,8.26,0.11,8.13,0.11,8s0.05-0.26,0.15-0.35l4.83-4.84c0.2-0.19,0.51-0.19,0.71,0 c0.19,0.2,0.19,0.51,0,0.71L1.81,7.5h13.58c0.27,0,0.5,0.22,0.5,0.5s-0.23,0.5-0.5,0.5H1.81l3.99,3.98c0.19,0.19,0.19,0.51,0,0.71 C5.6,13.38,5.29,13.38,5.09,13.19z"
        />
      </g>
    </svg>
    Back
  </button>
  <section
    className="fsuQGr"
  >
    <div>
      <h5
        className="jIUOlj"
      >
        NAME
      </h5>
      <input
        className="denvxK"
        onChange={[Function]}
        onKeyDown={null}
        placeholder="Label name"
        type="text"
        value=""
      />
    </div>
  </section>
  <section
    className="fsuQGr"
  >
    <h5>
      SELECT COLOR: 
    </h5>
    <div
      className="iuHIwX"
    />
  </section>
  <section
    className="koQQzt"
  >
    <button
      className="iHaZMM"
      onClick={[Function]}
    >
      Save label
    </button>
  </section>
</div>
`;
